plaidml/keras/backend.py

```diff
@@ -1141,7 +1141,7 @@ def __init__(self, indices, num_classes):
             }}""".format(idim=', '.join(['I{}'.format(k) for k in range(indices.shape.ndims)]),
                          iidx=', '.join(['i{}'.format(k) for k in range(indices.shape.ndims)]))
 
-        outshape = ptile.Shape(plaidml.DType.BOOLEAN,
+        outshape = ptile.Shape(plaidml.DType.FLOAT32,
```

Comment:
As a part of Keras, this might be better as
```
outshape = ptile.convert_np_dtype_to_pml(floatx())
```
This would make onehot return the user-specified default floating point type rather than fp32 regardless of what floating point type the user asked to be default.

Reply:
Sure... makes perfect sense.